--- libgnucash/backend/sql/gnc-transaction-sql.cpp
+++ libgnucash/backend/sql/gnc-transaction-sql.cpp
@@ -524,11 +524,19 @@
     }
     split_info.be = sql_be;
     split_info.is_ok = TRUE;

-    g_list_foreach (xaccTransGetSplitList (pTx), delete_split_slots_cb,
-                    &split_info);
+    std::vector<const GncGUID*> split_guids;
+    for (GList* node = xaccTransGetSplitList (pTx); node; node = node->next)
+    {
+        Split* pSplit = GNC_SPLIT (node->data);
+        if (pSplit) {
+            split_guids.push_back(qof_instance_get_guid(QOF_INSTANCE(pSplit)));
+        }
+    }
+
+    split_info.is_ok = gnc_sql_slots_delete_multiple(sql_be, split_guids);

     return split_info.is_ok;
 }
